### PR TITLE
Closes #2537:  `array_view_test` conversion for new framework

### LINKED ISSUE #2537: `array_view_test.py` conversion for new test framework
Convert `array_view_test.py` to work in the new test framework. The existing file should be maintained since we will be doing a complete swap once all tests are converted. Only the existing tests need to be converted, there should not be any additional testing that needs to be added.

### PR DESCRIPTION
**Pull request by jaketrookman**
This PR (Closes #2537) updates  `array_view_test` to the new testing framework.

Updated to be parameterized for different dimensioned arrays in a variety of scale.

### CODE REVIEW

File: tests/array_view_test.py [hunk 1]
```diff
@@ -93,17 +93,21 @@ def test_get_bool_pdarray(self):
...
         a[False, True, True] = 5
         self.assertListEqual(n.tolist(), a.to_list())
+        # print(n)
         n[True, False, True] = 6
+        # print(n)
         a[True, False, True] = 6
         self.assertListEqual(n.tolist(), a.to_list())
+        # print(n)
         n[True, True, False] = 13
+        # print(n)
```
Comment (jaketrookman) [inline tests/array_view_test.py:110]: Forgot to remove these print statements

File: PROTO_tests/tests/array_view_test.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+SHAPE = {
+    4: (2, 2),
+    30: (5, 3, 2),
+    252: (4, 9, 7),
+    693: (3, 7, 3, 11),
+}
+
+SIZE = [4, 30, 252, 693]
+TRUTH = [(True, True, True), (False, True, True), (True, False, True), (True, True, False)]
```
Comment (TheOddCzar789) [inline PROTO_tests/tests/array_view_test.py:16]: This isn't used anywhere

File: PROTO_tests/tests/array_view_test.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
+from itertools import product
+
+import numpy as np
+import arkouda as ak
+import pytest
```
[the remark this answers is not in the document]
Comment (jaketrookman) [reply PROTO_tests/tests/array_view_test.py:4]: Updated

File: PROTO_tests/tests/array_view_test.py (hunk 2 above)
Comment (jaketrookman) [reply PROTO_tests/tests/array_view_test.py]: Updated

**Review by Ethan-DeBandi99 — APPROVED** [review]

File: PROTO_tests/tests/array_view_test.py
Comment (pierce314159) [inline PROTO_tests/tests/array_view_test.py:1]: seems like we need to run `black` again on this file

File: PROTO_tests/tests/array_view_test.py [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        assert np.array_equal(uav[iind], uav[uind])
+
+    @pytest.mark.parametrize("size", SIZE)
+    def test_set_index(self, size):
+        inav = np.arange(size).reshape(SHAPE[size])
+        unav = np.arange(size, dtype=np.uint64).reshape(SHAPE[size])
+        iav = ak.arange(size).reshape(SHAPE[size])
+        uav = ak.arange(size, dtype=ak.uint64).reshape(SHAPE[size])
+
+        nind = (1,) * inav.ndim
```
Comment (jaketrookman) [inline PROTO_tests/tests/array_view_test.py:76]: Extra close review here about what values should be populating this

File: PROTO_tests/tests/array_view_test.py [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+            # cannot reshape array of size 9 into shape (4,3)
+            ak.arange(9).reshape(4, 3)
+
+    @pytest.mark.parametrize("size", SIZE)
+    def test_int_list_indexing(self, size):
+        N = len(SHAPE[size])
+        iav = ak.arange(size).reshape(SHAPE[size])
+        uav = ak.arange(size, dtype=np.uint64).reshape(SHAPE[size])
+
+        iind = ak.ones(N, dtype=ak.uint64)
```
Comment (pierce314159) [inline PROTO_tests/tests/array_view_test.py:64]: we define `N` but only use it once

File: PROTO_tests/tests/array_view_test.py [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+        uind = ak.cast(iind, np.uint64)
```
Comment (pierce314159) [inline PROTO_tests/tests/array_view_test.py:65]: I think one of these (`iind`) is meant to have integer dtype, but they're both `uint` here

File: PROTO_tests/tests/array_view_test.py [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+            n = np.arange(size).reshape(SHAPE[size])
+            a = ak.arange(size).reshape(SHAPE[size])
+            truth = [True] * N
+            if i > 0:
+                truth[i - 1] = False
+            truth = tuple(truth)
+            n_bool_list = n[truth].tolist()
+            a_bool_list = a[truth].to_list()
+
+            assert n_bool_list == a_bool_list
```
Comment (pierce314159) [inline PROTO_tests/tests/array_view_test.py:103]: We don't need to redefine `n` and `a` every time and also there might be a more direct to compute `truth`
```suggestion
        n = np.arange(size).reshape(SHAPE[size])
        a = ak.arange(size).reshape(SHAPE[size])
        for i in range(0, N+1):
            truth = tuple(j != i-1 if i > 0 else True for j in range(N))
            n_bool_list = n[truth].tolist()
            a_bool_list = a[truth].to_list()

            assert n_bool_list == a_bool_list
```

File: PROTO_tests/tests/array_view_test.py [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+        for i in range(0, N+1):
+            n = np.arange(size).reshape(SHAPE[size])
+            a = ak.arange(size).reshape(SHAPE[size])
+            truth = [True] * N
+            if i > 0:
+                truth[i-1] = False
+            truth = tuple(truth)
+            n[truth] = 7
+            a[truth] = 7
+            assert n.tolist() == a.to_list()
```
Comment (pierce314159) [inline PROTO_tests/tests/array_view_test.py:118]: same suggestions for moving `n` and `a` out of loop and constructing truth using a tuple comprehension

**Review by pierce314159 — COMMENTED** [review]
Not a ton to comment on, this seems pretty close. Good work!

File: PROTO_tests/tests/array_view_test.py (hunk 4 above)
Comment (pierce314159) [reply PROTO_tests/tests/array_view_test.py:76]: Yah i think all 1's should be fine, but if we wanted to do random indexing we could do something like this
```python
import random
tuple(random.randint(0,y-1) for y in shape)
```

File: PROTO_tests/tests/array_view_test.py (hunk 8 above)
Comment (jaketrookman) [reply PROTO_tests/tests/array_view_test.py]: Updated `truth` construction, but keeping `n` and `a` inside the loop to regenerate between each test. Also added a random value to test each time

File: PROTO_tests/tests/array_view_test.py (hunk 7 above)
Comment (jaketrookman) [reply PROTO_tests/tests/array_view_test.py]: Updated

File: PROTO_tests/tests/array_view_test.py (hunk 6 above)
Comment (jaketrookman) [reply PROTO_tests/tests/array_view_test.py]: Updated

File: PROTO_tests/tests/array_view_test.py (hunk 5 above)
Comment (jaketrookman) [reply PROTO_tests/tests/array_view_test.py]: Updated

File: PROTO_tests/tests/array_view_test.py
Comment (jaketrookman) [reply PROTO_tests/tests/array_view_test.py:1]: Updated

File: PROTO_tests/tests/array_view_test.py [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+import numpy as np
+import pytest
+
+import arkouda as ak
+
+SHAPE = {
+    4: (2, 2),
+    30: (5, 3, 2),
+    252: (4, 9, 7),
+    693: (3, 7, 3, 11),
```
Comment (pierce314159) [inline PROTO_tests/tests/array_view_test.py:13]: I think we should lower these numbers especially for the ones that have more dimensions. For `test_reshape_order` we do a distinct index for each combination and that can take a while with multi-locale. 
so I would shift the larger dimension sizes to the ones with few dimensions

Comment (jaketrookman) [reply PROTO_tests/tests/array_view_test.py]: Updated

**Review by Ethan-DeBandi99 — APPROVED** [review]

**Pull request merged**